File: giftastic-deployment-env/src/main/java/com/giftastic/giftastic/modules/report/service/ReportService.java
package com.giftastic.giftastic.modules.report.service;

import java.util.UUID;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.giftastic.giftastic.modules.report.domain.Report;
import com.giftastic.giftastic.modules.report.domain.ReportStatus;
import com.giftastic.giftastic.modules.report.domain.ReportType;

public interface ReportService {
    Report createReport(UUID reporterId, ReportType reportType, UUID reportedEntityId, String reason, String description);
    Report getReportById(UUID reportId);
    Page<Report> getAllReports(Pageable pageable);
    Page<Report> getReportsByStatus(ReportStatus status, Pageable pageable);
    Page<Report> getReportsByType(ReportType reportType, Pageable pageable);
    Page<Report> getReportsByEntity(UUID entityId, Pageable pageable);
    Page<Report> getMyReports(UUID reporterId, Pageable pageable);
    void markUnderReview(UUID reportId, UUID adminId);
    void markActionTaken(UUID reportId, UUID adminId, String notes, String action);
    void dismissReport(UUID reportId, UUID adminId, String notes);
    void resolveReport(UUID reportId, UUID adminId, String notes);
}

File: giftastic-frontend-deploy/frontend/src/services/reportService.js
import api from './api';

export const reportService = {
  // User endpoints
  createReport: async (reportData) => {
    const response = await api.post('/reports', reportData);
    return response.data;
  },

  getMyReports: async (page = 0, size = 20) => {
    const response = await api.get('/reports/my-reports', {
      params: { page, size }
    });
    return response.data;
  },

  // Admin endpoints
  getAllReports: async (page = 0, size = 20) => {
    const response = await api.get('/reports', {
      params: { page, size }
    });
    return response.data;
  },

  getReportsByStatus: async (status, page = 0, size = 20) => {
    const response = await api.get(`/reports/status/${status}`, {
      params: { page, size }
    });
    return response.data;
  },

  getReportsByType: async (type, page = 0, size = 20) => {
    const response = await api.get(`/reports/type/${type}`, {
      params: { page, size }
    });
    return response.data;
  },

  getReportsByEntity: async (entityId, page = 0, size = 20) => {
    const response = await api.get(`/reports/entity/${entityId}`, {
      params: { page, size }
    });
    return response.data;
  },

  getReport: async (reportId) => {
    const response = await api.get(`/reports/${reportId}`);
    return response.data;
  },

  markUnderReview: async (reportId) => {
    const response = await api.patch(`/reports/${reportId}/under-review`);
    return response.data;
  },

  markActionTaken: async (reportId, notes, action) => {
    const response = await api.patch(`/reports/${reportId}/action-taken`, null, {
      params: { notes, action }
    });
    return response.data;
  },

  dismissReport: async (reportId, notes) => {
    const response = await api.patch(`/reports/${reportId}/dismiss`, null, {
      params: { notes }
    });
    return response.data;
  },

  resolveReport: async (reportId, notes) => {
    const response = await api.patch(`/reports/${reportId}/resolve`, null, {
      params: { notes }
    });
    return response.data;
  }
};
